### DISCUSSION TITLE
A warning about the "removal of transient, non-exclusive queues" in the logs

### DISCUSSION THREAD

**Discussion by LionelVallet** [rabbitmq/rabbitmq-delayed-message-exchange#297] [Q&A] [answered]
Version 4 has deprecated this feature. The [documentation](https://www.rabbitmq.com/blog/2021/08/21/4.0-deprecation-announcements#removal-of-transient-non-exclusive-queues) states:

> ## Removal of transient, non-exclusive queues
> 
> ### Why are we making this decision?
> Transient queues are queues whose lifetime is linked to the uptime of the node they are declared on. In a single node cluster they are removed when the node is restarted. In a clustered environment they are removed when the node they are hosted on is restarted.
> 
> Correct use of transient queues requires that an application developer knows something about node uptime. Further, a node restart isn't a good way to remove unused queues.
> 
> There is one kind of transient queue that is not included in this deprecation; the exclusive queue. Exclusive queues are linked to the lifetime of the declaring connection which is something that an application developer can take into account and leverage.
> 
> By deprecating transient queues we are removing a potentially confusing queue option. We also reduce pressure on the boot procedure, as transient queues are currently removed on boot.
> 
> ### What alternatives exist?
> Queue TTL should be used for auto-deleting unused, idle queues after some time of inactivity.
> 
> Exclusive queues: these are deleted once all connections to the queue are removed.

**Comment by michaelklishin** [comment, marked as answer]
This plugin is not responsible for the log message in question. Your apps declare the queues, not this plugin.

Most likely you are looking at something similar to what https://github.com/rabbitmq/rabbitmq-server/pull/12734, https://github.com/rabbitmq/rabbitmq-server/pull/12738 seek to address.

**Reply by LionelVallet** [reply]
You're right, this message was intended for MassTransit and I got confused at the last moment. I'm very sorry. Thank you for your reply.

**Reply by michaelklishin** [reply]
Note that there is no clear removal date for transient, non-exclusive queues. So this is not an urgent matter for libraries such as MassTransit to address. As long as they support quorum queues (and perhaps streams) as options and do not use classic mirrored queues, they have months if not a good year to figure the transition.

**Reply by phatboyg** [reply]
Quorum queues can already be declared, even for the bus queue which has an expiration so that it is automatically removed after the idle period.
